lua_configuration/lua.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/openkruise/rollouts/api/v1alpha1"
+	"github.com/openkruise/rollouts/pkg/trafficrouting/network/custom"
+	"github.com/openkruise/rollouts/pkg/util/luamanager"
+	lua "github.com/yuin/gopher-lua"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+
+	utilpointer "k8s.io/utils/pointer"
+)
+
+type TestCase struct {
+	Rollout        *v1alpha1.Rollout            `json:"rollout,omitempty"`
+	TrafficRouting *v1alpha1.TrafficRouting     `json:"trafficRouting,omitempty"`
+	Original       *unstructured.Unstructured   `json:"original,omitempty"`
+	Expected       []*unstructured.Unstructured `json:"expected,omitempty"`
+}
+
+// convert testdata to lua object for debugging
+func main() {
+	err := PathWalk()
+	if err != nil {
+		fmt.Println(err)
+	}
+}
+
+func PathWalk() error {
+	err := filepath.Walk("./", func(path string, f os.FileInfo, err error) error {
+		if !strings.Contains(path, "trafficRouting.lua") {
+			return nil
+		}
+		if err != nil {
+			return fmt.Errorf("failed to walk path: %s", err.Error())
+		}
+		dir := filepath.Dir(path)
+		if _, err := os.Stat(filepath.Join(dir, "testdata")); err != nil {
```

Comment:
is testdata directory included in this PR?

Reply:
testdata and lua scripts are not included in this PR. This function aims to convert testdata into lua objects. When the user wants to debug their lua scripts, they could use this function to generate lua objects and copy the generated objects to their lua scripts for debugging.